### ISSUE TITLE
-p argument not parsing?

### ISSUE THREAD

**Issue by brandonros** [philrandal/gpsctl#16]
```
$ ./gpsctl -p /dev/cu.usbmodemTAhJALxt1
Error: device "/dev/serial0" is not a terminal
```

**Comment by philrandal** [comment]
I can't reproduce this.

My last patch re -p parsing was

https://github.com/philrandal/gpsctl/commit/b31a86a1184c941e1d658a228e82a0800668a7c8#diff-d1c108c7a2cc401ad54b352eeb2927d0ec98063acd11b5ec8cd7fdfbc36cea36

which fixed it for me in my tests

**Comment by philrandal** [comment]
We read gpsctl:port from gpsctl.conf before parsing the commandline parameters (gpsctl.c line 1395).

The code might need refactoring to process the commandline first and ignore the config file value if -p is specified.

The workaround is to specify the serial port in gpsctl.conf instead of via -p on the commandline.

I'll take a look at this in the next day or two.

**Comment by philrandal** [comment]
Fixed in gpsctl v1.16
